pkg/usecases/authentication/device_code/device_code.go

```diff
@@ -0,0 +1,67 @@
+package device_code
```

Comment:
Nits, it is recommended to use a package name without `under_scores`.
https://go.dev/blog/package-names

```suggestion
package devicecode
```